query-engine/core/src/interactive_transactions/actor_manager.rs

```diff
@@ -81,6 +90,7 @@ impl TransactionActorManager {
         operation: Operation,
         trace_id: Option<String>,
     ) -> crate::Result<ResponseData> {
+        self.handle_closed_tx(tx_id, "query").await?;
```

Comment:
I think this check should be moved into the `get_client`. By checking it here means we have to wait for a read lock every time we do a query. If we put in the `get_client` when we cannot find the itx in the clients list means we will check this a lot less often. In theory, we should never have to check this LRU because the majority of the time the user will be doing the itx in the correct order.